samcli/lib/build/app_builder.py

```diff
@@ -430,6 +430,11 @@ def _build_lambda_image(self, function_name: str, metadata: Dict, architecture:
             LOG.debug("%s image is built for %s function", build_image, function_name)
         except docker.errors.BuildError as ex:
             LOG.error("Failed building function %s", function_name)
+            for log in ex.build_log:
+                if "stream" in log:
+                    self._stream_writer.write_str(log["stream"])
```

Comment:
When writing to `_stream_writer`, I think checking if it is clear what type of output will be shown. I think we should include context to indicate whether it is part of a standard log or an error message, such as prepending "[Error]: " when writing log["error"].


```
except docker.errors.BuildError as ex:
    LOG.error("Failed building function %s", function_name)
    for log in ex.build_log:
        if "stream" in log:
            self._stream_writer.write_str(f"Log: {log['stream']}")
        elif "error" in log:
            self._stream_writer.write_str(f"[Error]: {log['error']}")
        else:
            LOG.warning("Unexpected log entry format: %s", log)

```

Reply:
We should consider a way to differentiate between different types of docker.errors.BuildError logs (e.g., specific errors vs. general stream logs), more granular handling might be helpful. This could mean adding conditionals or further parsing within the loop. I was thinking